kmath-geometry/src/commonMain/kotlin/scientifik/kmath/geometry/EuclideanSpace.kt

```diff
@@ -0,0 +1,57 @@
+package scientifik.kmath.geometry
+
+import scientifik.kmath.dimensions.*
+import scientifik.kmath.operations.ExtendedField
+import scientifik.kmath.operations.Field
+import scientifik.kmath.operations.invoke
+import scientifik.kmath.structures.asSequence
+
+internal data class VectorImpl<T, D : Dimension>(private val array: List<T>) : Vector<T, D> {
+    override val size: Int get() = array.size
+
+    override fun get(index: Int): T = when {
+        index < 0 || index >= size -> error("Accessing outside of point bounds")
+        else -> array[index]
+    }
+
+    override fun iterator(): Iterator<T> = array.iterator()
+}
+
+fun <T> vectorOf(a1: T): Vector<T, D1> = VectorImpl(listOf(a1))
+fun <T> vectorOf(a1: T, a2: T): Vector<T, D2> = VectorImpl(listOf(a1, a2))
+fun <T> vectorOf(a1: T, a2: T, a3: T): Vector<T, D3> = VectorImpl(listOf(a1, a2, a3))
+fun <T> vectorOf(a1: T, a2: T, a3: T, a4: T): Vector<T, D4> = VectorImpl(listOf(a1, a2, a3, a4))
+fun <T> vectorOf(a1: T, a2: T, a3: T, a4: T, a5: T): Vector<T, D5> = VectorImpl(listOf(a1, a2, a3, a4, a5))
+
+private fun <T> Field<T>.square(a: T): T = a * a
```

Comment:
Rename to `sqr` and change the receiver type to `Ring<T>` :)

Reply:
By the way, maybe a function like this should be added to the core. 

upd: It already is, but it is targeted to `MathElement` not to `Algebra`